### DISCUSSION TITLE
How to see values in a friendly view inside JSON from debugger?

### DISCUSSION THREAD

**Discussion by zliucd** [nlohmann/json#3773] [General]
Hello,

How to see values in a friendly view inside JSON from debugger(gdb or IDE)? Currently, I can only see internal structures defined in JSON, e.g, ```m_type```, ```m_value```, which is hard to read.

Thanks.

**Comment by falbrechtskirchinger** [comment]
https://github.com/nlohmann/json/tree/develop/tools/gdb_pretty_printer

**Reply by zliucd** [reply]
Thanks. Is that possible to view user-friendly contents, e.g., key-value pair, directly from IDE (in the future)?

**Reply by falbrechtskirchinger** [reply]
Depends on your IDE. If your IDE is using gdb the gdb_pretty_printer should work just fine. It does in VSCode, for example.

![image](https://user-images.githubusercontent.com/320854/192973146-18c6de6f-4108-4b90-9554-c6c58fc93019.png)

There's also a Natvis file for tools that support it: https://github.com/nlohmann/json/blob/develop/nlohmann_json.natvis
